autopush-common/src/reliability.rs

```diff
@@ -46,18 +45,36 @@ const CONNECTION_EXPIRATION: TimeDelta = TimeDelta::seconds(10);
 #[serde(rename_all = "snake_case")]
 #[strum(serialize_all = "snake_case")]
 pub enum ReliabilityState {
-    Received,        // Message was received by the Push Server
-    Stored,          // Message was stored because it could not be delivered immediately
-    Retrieved,       // Message was taken from storage for delivery
-    IntTransmitted,  // Message was handed off between autoendpoint and autoconnect
-    IntAccepted,     // Message was accepted by autoconnect from autopendpoint
-    Transmitted,     // Message was handed off for delivery to the UA
-    Accepted,        // Message was accepted for delivery by the UA
-    Delivered,       // Message was provided to the WebApp recipient by the UA
-    DecryptionError, // Message was provided to the UA and it reported a decryption error
-    NotDelivered,    // Message was provided to the UA and it reported a not delivered error
-    Expired,         // Message expired naturally (e.g. TTL=0)
-    Errored,         // Message resulted in an Error state and failed to be delivered.
+    Received,          // Message was received by the Push Server
```

Comment:
I would love these to be doc-comments (`///`) above each variant. And the variant itself. Once we generate docs, we can read through them :)

Reply:
https://mozilla-hub.atlassian.net/browse/PUSH-534